### PR TITLE
feat(map_based_prediction): upgrade yaw difference tolerance

### PR DESCRIPTION
**Pull request by purewater0901**
Signed-off-by: yutaka <purewater0901@gmail.com>

## Description
When yaw inversion occurred, map based prediction does not generate appropriate paths for that object even the object is on the reasonable lanelet. In this PR, I fixed that problem, and allow map based prediction to generate predicted trajectories when the object's yaw is inverse's 180. 

Before the PR:
image

After the PR:
image

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by yukkysaito — APPROVED** [review]
LGTM

File: perception/map_based_prediction/src/map_based_prediction_node.cpp [hunk 1]
```diff
@@ -423,7 +423,8 @@ bool MapBasedPredictionNode::checkCloseLaneletCondition(
   // of the lanelets that are below max_dist and max_delta_yaw
   if (
     lanelet.first < dist_threshold_for_searching_lanelet_ &&
-    abs_norm_delta < delta_yaw_threshold_for_searching_lanelet_) {
+    (abs_norm_delta < delta_yaw_threshold_for_searching_lanelet_ ||
+     abs_norm_delta > M_PI - delta_yaw_threshold_for_searching_lanelet_)) {
```
Comment (yukkysaito) [inline perception/map_based_prediction/src/map_based_prediction_node.cpp:427]:
```suggestion
    (delta_yaw_threshold_for_searching_lanelet_< abs_norm_delta  ||
     abs_norm_delta < delta_yaw_threshold_for_searching_lanelet_ )) {
```
This one might be better. nitpick comment 🙏

Comment (purewater0901) [reply perception/map_based_prediction/src/map_based_prediction_node.cpp:427]: @yukkysaito Is it okay to remove M_PI from the condition?

Comment (yukkysaito) [reply perception/map_based_prediction/src/map_based_prediction_node.cpp:427]: sorry, it's misstake
```suggestion
    (M_PI  - delta_yaw_threshold_for_searching_lanelet_< abs_norm_delta  ||
     abs_norm_delta < delta_yaw_threshold_for_searching_lanelet_)) {
```

Comment (purewater0901) [reply perception/map_based_prediction/src/map_based_prediction_node.cpp]: done

**Review by yukkysaito — DISMISSED** [review]

**Comment by purewater0901** [comment]
@yukkysaito Since I had the format error, could you approve this PR after all the CI has passed ?

**Review by yukkysaito — APPROVED** [review]

**Comment by tkimura4** [comment]
FYI:
The behavior of a vehicle running backwards in the lane seems strange
image

**Pull request merged**